packages/core-p2p/lib/guard.js

```diff
@@ -0,0 +1,113 @@
+'use strict'
+
+const moment = require('moment')
+const semver = require('semver')
+const logger = require('@arkecosystem/core-container').resolvePlugin('logger')
+const isMyself = require('./utils/is-myself')
+
+class Guard {
+  constructor () {
+    this.suspensions = {}
+  }
+
+  init (config) {
+    this.config = config
+
+    return this
+  }
+
+  /**
+   * Get a list of all suspended peers.
+   * @return {Object}
+   */
+  all () {
+    return this.suspensions
+  }
+
+  /**
+   * Get the suspended peer for the give IP.
+   * @return {Object}
+   */
+  get (ip) {
+    return this.suspensions[ip]
+  }
+
+  /**
+   * Suspends a peer unless whitelisted.
+   * @param {Peer} peer
+   */
+  suspend (peer) {
+    if (this.config.peers.whiteList && this.config.peers.whiteList.includes(peer.ip)) {
+      return
+    }
+
+    const until = moment().add(this.manager.config.suspendMinutes, 'minutes')
+
+    this.suspensions[peer.ip] = {
+      peer,
+      until,
+      untilHuman: until.format('h [hrs], m [min]')
+    }
+
+    delete this.peers[peer.ip]
```

Comment:
This has no effect here. The `peers` only exist in `Monitor`.

Reply:
:ok_hand: thanks